src/components/LineAreaGraph.tsx

```diff
@@ -0,0 +1,249 @@
+
+type TooltipStyleArgs = {
+  series: number[][]
+  seriesIndex: number
+  dataPointIndex: number
+}
+
+interface Props {
+  series: number[]
+  categories: (number | string)[]
+  yAxisType: 'plain' | 'tx'
+  xAxisType: 'datetime' | 'numeric' | 'category'
+  isLoading: boolean
+}
+
+const formatYAxis =
+  (type: Props['yAxisType']) =>
+  (value: number): string => {
+    if (type === 'tx') {
+      const formattedParts = formatNumberForDisplay(value, 'quantity')
+      const integer = formattedParts[0]
+      const suffix = formattedParts[2]
+      return (integer && integer + suffix) || ''
+    }
+    return value.toString()
+  }
+
+const formatXAxis =
+  (type: Props['xAxisType']) =>
+  (value: string | string[]): string => {
+    const _value = Array.isArray(value) ? value[0] : value
+    if (type === 'datetime') {
+      return formatToMonthDay(new Date(_value))
+    }
+    return _value
+  }
+
+function formatSeriesNumber(type: Props['yAxisType'], value: number): string {
+  // TODO: Special formatting (TX/s, etc)
+  if (type === 'tx') {
+    return value.toString()
+  }
+  return value.toString()
+}
+
+const fontSizeInPx = 12
+
+function getOffsetXYAxisLabel(series: number[], type: Props['yAxisType']): number {
+  const largestNumber = series.reduce((l, c) => Math.max(l, c), 0)
+  const formatted = formatYAxis(type)(largestNumber)
+  return formatted.length * fontSizeInPx
+}
+
+function formatToMonthDay(dt: Date): string {
+  const month = dt.getMonth().toString().padStart(2, '0')
+  const day = dt.getDay().toString().padStart(2, '0')
+  return month + '/' + day
+}
+
+function formatToYearMonthDay(dt: Date): string {
+  const year = dt.getFullYear()
+  return year + '/' + formatToMonthDay(dt)
+}
+
+const LineAreaGraph = ({ series, categories, xAxisType, yAxisType, isLoading }: Props) => {
+  const theme = useTheme()
+  const options = {
+    chart: {
+      toolbar: {
+        show: false
+      }
+    },
+    xaxis: {
+      type: xAxisType,
+      categories,
+      axisTicks: {
+        color: theme.borderSecondary
+      },
+      axisBorder: {
+        show: false
+      },
+      labels: {
+        style: {
+          colors: theme.textSecondary
+        },
+        formatter: formatXAxis(xAxisType)
+      },
+      tooltip: {
+        enabled: false
+      }
+    },
+    yaxis: {
+      floating: true,
+      labels: {
+        style: {
+          colors: theme.textSecondary,
+          fontSize: fontSizeInPx + 'px'
+        },
+        offsetY: -8,
+        offsetX: getOffsetXYAxisLabel(series, yAxisType),
+        formatter: formatYAxis(yAxisType)
+      }
+    },
+    grid: {
+      borderColor: theme.borderPrimary,
+      padding: {
+        top: 0,
+        right: 0,
+        bottom: 0,
+        left: 2
+      }
+    },
+    markers: {
+      colors: ['#FFFFFF']
+    },
+    tooltip: {
+      theme: false as unknown as string,
+      custom: function ({ series, seriesIndex, dataPointIndex }: TooltipStyleArgs) {
```

Comment:
nit: let's use an arrow function

Reply:
I think it's better to use the following:

```
{
  theme: ...,
  custom(...) { }
}
```